CONTRIBUTING.md

```diff
@@ -45,17 +45,23 @@ Additionally, please update [the changelog](CHANGELOG.md) if you're making any u
 
 ## Working on the Code
 
-If you wish to work on the OpenTofu CLI source code, you'll first need to install the [Go](https://golang.org/) compiler and the version control system [Git](https://git-scm.com/).
+If you wish to work on the OpenTofu CLI source code, you'll first need to install the [Git](https://git-scm.com/) version control system. Use Git to clone this repository into a location of your choice. OpenTofu uses [Go Modules](https://blog.golang.org/using-go-modules), and so you should *not* clone it inside your `GOPATH`.
 
-At this time the OpenTofu development environment is targeting only Linux and Mac OS X systems. While OpenTofu itself is compatible with Windows, unfortunately the unit test suite currently contains Unix-specific assumptions around maximum path lengths, path separators, etc.
+After that, you can either install the [Go](https://golang.org/) compiler locally, or use [Docker](https://www.docker.com/) to build in a container. At this time the OpenTofu development environment targets only Linux and MacOS systems. While OpenTofu itself is compatible with Windows, unfortunately the unit test suite currently contains Unix-specific assumptions around maximum path lengths, path separators, etc. This means that using Docker is the best option if working on Windows.
+
+If using Visual Studio Code or IntelliJ, a [devcontainer](.devcontainer.json) is included which integrates directly with Docker. Simply open the repository and you should be prompted to re-open in the container. At this point you can proceed as if [building natively](#building-natively) on Linux.
```

Comment:
Goland would be more appropriate to mention here since it's a Go project. Also, the instructions seem to be inaccurate since Goland doesn't prompt to open the dev container, only VSCode does that. You need to open the `.devcontainer.json` file and click the little box icon:

![image](https://github.com/opentofu/opentofu/assets/86970079/4a042353-a322-4e8f-903f-b38882499bfb)

For VSCode, the devcontainers addon needs to be installed explicitly as it is not bundled.

Reply:
If you have the Remote Containers extension installed, you should get a toast popup as soon as you open the repository: 
![image](https://github.com/opentofu/opentofu/assets/3674196/15df92c3-a029-4868-989c-504f544a26a0)

I've updated the wording to accommodate the above points. Thanks!